### PR TITLE
Using UUIDs in URLs for users, publishers, importers, distros, and workers

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #3125
https://pulp.plan.io/issues/3125

### CODE REVIEW

File: pulpcore/pulpcore/app/serializers/task.py [hunk 1]
```diff
@@ -95,8 +94,7 @@ class Meta:
 
 class WorkerSerializer(ModelSerializer):
     _href = serializers.HyperlinkedIdentityField(
-        view_name='workers-detail',
-        lookup_field='name'
+        view_name='workers-detail'
```
Comment (asmacdo) [inline pulpcore/pulpcore/app/serializers/task.py]: (nit) I'd prefer this on 1 line

File: docs/contributing/3.0-development/rest-api.rst [hunk 2]
```diff
@@ -198,37 +197,6 @@ name of that model when using a related field.
 This is enough of a tricky problem that it has its own section in the docs a little bit below,
 called "Master/Detail Relationships Overview".
 
-lookup_field annoyances
```
Comment (asmacdo) [inline docs/contributing/3.0-development/rest-api.rst]: Thanks for remembering this doc.

File: docs/contributing/3.0-development/rest-api.rst [hunk 3]
```diff
@@ -240,7 +208,7 @@ explicit Serializer class, DRF provides an excellent bit of functionality::
 	>>> from serializers import RepositorySerializer
 	>>> RepositorySerializer()
 	RepositorySerializer():
-		_href = HyperlinkedIdentityField(lookup_field='name', view_name='repository-detail')
+		_href = HyperlinkedIdentityField(view_name='repository-detail')
```
Comment (asmacdo) [inline docs/contributing/3.0-development/rest-api.rst]: s/repository-detail/repositories-detail/

File: docs/contributing/3.0-development/rest-api.rst [hunk 4]
```diff
@@ -416,7 +383,6 @@ that allows the same request:
             fields = ['name']
 
     class RepositoryViewSet(viewsets.ModelViewSet):
-        lookup_field = 'name'
 		queryset = models.Repository.objects.all()
 		serializer_class = serializers.RepositorySerializer
 		filter_class = RepositoryFilter
```
Comment (asmacdo) [inline docs/contributing/3.0-development/rest-api.rst]: While you're in here, would you fix this indentation?

File: pulpcore/pulpcore/app/viewsets/repository.py [hunk 5]
```diff
@@ -115,12 +115,11 @@ class Meta:
 
 class ImporterViewSet(NamedModelViewSet):
     endpoint_name = 'importers'
-    lookup_field = 'name'
     serializer_class = ImporterSerializer
     queryset = Importer.objects.all()
     filter_class = ImporterFilter
 
-    def update(self, request, name, partial=False):
+    def update(self, request, pk, partial=False):
```
Comment (asmacdo) [inline pulpcore/pulpcore/app/viewsets/repository.py]: It's sort of weird that the pk gets passed everywhere, but importer.pk is what is used. I don't think we should change anything, this isn't relevant to your work, but your work does highlight this weirdness.

Comment (dkliban) [reply pulpcore/pulpcore/app/viewsets/repository.py]: This code will change when my DRY up PR is merged. I am rebasing it right now.

File: docs/contributing/3.0-development/rest-api.rst [hunk 6]
```diff
@@ -80,10 +80,8 @@ Serializer Notes
   be explicit.
 
 * Serialized objects will provide thier own URL as the value of the "_href" field on the serializer.
-  If your API endpoint does not use the primary key as its lookup field (such as repository being
-  looked up by name), you will need to use a ``rest_framework.serializers.HyperlinkedIdentifyField``
-  to generate the ``_href`` field value correctly, by specifying its ``lookup_field`` and
-  ``view_name``.
+  You will need to use a ``rest_framework.serializers.HyperlinkedIdentifyField`` to generate the
+  ``_href`` field value correctly, by specifying its ``lookup_field`` and ``view_name``.
```
Comment (asmacdo) [inline docs/contributing/3.0-development/rest-api.rst]: by specifying its ``view_name``. If this object is referenced in the url by a field other than the pk, you will also need to specify a ``lookup_field``

File: pulpcore/pulpcore/app/viewsets/repository.py (hunk 5 above)
Comment (asmacdo) [reply pulpcore/pulpcore/app/viewsets/repository.py]: thanks @dkliban

File: docs/contributing/3.0-development/rest-api.rst (hunk 6 above)
Comment (asmacdo) [reply docs/contributing/3.0-development/rest-api.rst]: also s/correctly//

**Pull request merged**